packages/stex-react-renderer/src/lib/ConceptHistoryTable.tsx

```diff
@@ -31,49 +59,136 @@ function ConceptHistoryTable({
 }) {
   const [conceptHistoryData, setConceptHistoryData] =
     useState<ConceptHistory>();
+  const [loading, setLoading] = useState<boolean>(false);
 
   useEffect(() => {
-    getConceptHistory(concept).then(setConceptHistoryData);
+    setConceptHistoryData(undefined);
+    setLoading(true);
+    function fetchConceptHistory() {
+      getConceptHistory(concept)
+        .then(setConceptHistoryData)
+        .finally(() => setLoading(false));
+    }
+    fetchConceptHistory();
   }, [concept]);
 
+  let previousRemember = 0;
```

Comment:
Doing this directly in the component hurts readability. Move this code (computing `competenciesByDate` out of the component into one or more helper functions